tests/testthat/test-filters.R

```diff
@@ -71,6 +72,24 @@ with_mock_crunch({
         expect_no_request(is.public(f) <- FALSE)
     })
 
+    test_that("can get and set the team that a filter is on.", {
+        expect_identical(team(f), getTeams()[["Alpha Team"]])
+        expect_null(team(f_2))
+        
+        expect_PATCH(
+            team(f_2) <- getTeams()[["Alpha Team"]],
+            "https://app.crunch.io/api/datasets/1/filters/filter2/",
+            '{"team":"https://app.crunch.io/api/teams/team1/"}'
+        )
+        
+        # can also just use a url
+        expect_PATCH(
+            team(f_2) <- "https://app.crunch.io/api/teams/team1/",
+            "https://app.crunch.io/api/datasets/1/filters/filter2/",
+            '{"team":"https://app.crunch.io/api/teams/team1/"}'
+        )
+    })
+    
```

Comment:
* Add a test that no request is made if you set with the same thing
* Add a test of unsetting the team (sending NULL)
* Probably should have an integration test creating a filter, setting a team on it, and then setting NULL to remove it. I doubt we have e2e test coverage for this behavior anywhere else.

Reply:
Done, done, and done. (Also 😂to "I doubt we have e2e test coverage for this behavior anywhere else.")